pkg/restore/restore.go

```diff
@@ -1082,6 +1082,21 @@ func (ctx *restoreContext) getResource(groupResource schema.GroupResource, obj *
 	return u, nil
 }
 
+// getResource but with retry on retriable error
+func (ctx *restoreContext) getResourceWithRetries(groupResource schema.GroupResource, obj *unstructured.Unstructured, namespace, name string, retriable func(error) bool) (*unstructured.Unstructured, error) {
```

Comment:
Rather than repeating this whole func with a different Get call, maybe you could just add the retriable arg to `getResource` and then in that func, if it's nil, call `lister.Get`, and if it's not nil call `cache.GetRetriableWithCacheLister`.

This is a non-exported func only called in one place other than the one you're changing to add retries, so pass nil in for the other case, and retriable in where you're using the new func now.